File: First_Year/First Sem/PPS/Qn4.c
/*4. Write a C program to display the size of the data type int, char, float, double, long 
int and long double using sizeof( ) operator.  */
#include <stdio.h>

int main() {
    printf("%zu\n", sizeof(int));
    printf("%zu\n", sizeof(char));
    printf("%zu\n", sizeof(float));
    printf("%zu\n", sizeof(double));
    printf("%zu\n", sizeof(long int));
    printf("%zu\n", sizeof(long double));
    return 0;
}

File: First_Year/Second Sem/IOOP/Lab2/Qn4.java
/*4. Write a Java program to display non diagonal elements and find their sum. 
[Hint: Non Principal diagonal: The diagonal of a diagonal matrix from the top 
right to the bottom left corner is called non principal diagonal.]  */
import java.util.Scanner;

public class Qn4 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter matrix size: ");
        int n = sc.nextInt();

        int[][] a = new int[n][n];
        int sum = 0;
        
        System.out.print("Enter " + n*n + " Elements :");
        for (int i = 0; i < n; i++)
            for (int j = 0; j < n; j++)
                a[i][j] = sc.nextInt();

        System.out.print("Non-principal diagonal elements: ");
        for (int i = 0; i < n; i++) {
            System.out.print(a[i][n - i - 1] + " ");
            sum += a[i][n - i - 1];
        }

        System.out.println("\nSum = " + sum);

        sc.close();
    }
}
